File: WebShop_FSharp/Controllers/HomeController.fs
namespace WebShop_FSharp.Controllers

open System.Diagnostics
open DomainModels
open Microsoft.AspNetCore.Mvc
open Microsoft.Extensions.Logging
open WebShop_FSharp.ViewModels
open WebShop_FSharp.NullCoalescing

type HomeController(dbContext: ApplicationContext, logger: ILogger<HomeController>) =
    inherit Controller()
    
    member private this.dbContext = dbContext
    member private this.logger = logger
    
    
    member public this.Index = this.View
    
    member public this.Privacy = this.View
    
    [<ResponseCache(Duration = 0, Location = ResponseCacheLocation.None, NoStore = true)>]
    member public this.Error() =
        let activityCurrent = if Activity.Current = null then null else Activity.Current.Id
        let requestId = nullCoalesce activityCurrent this.HttpContext.TraceIdentifier
        this.View(ErrorViewModel(requestId))
    

File: WebShop_NULL/Controllers/HomeController.cs
using System.Diagnostics;
using DomainModels;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using WebShop_NULL.Models;
using WebShop_NULL.Models.ViewModels;
using WebShop_FSharp.ViewModels;

//F# version exists, currently doesn't work. TODO fix F# version
namespace WebShop_NULL.Controllers
{
    public class HomeController : Controller
    {
        private readonly ILogger<HomeController> _logger;

        public HomeController(ILogger<HomeController> logger, ApplicationContext dbContext)
        {
            _logger = logger;
        }

        public IActionResult Index()
        {
            return RedirectToAction("Index", "Catalog"); //Redirects to catalog
        }
    }
}